### PR TITLE
rpm: Upgrade to 4.18.0-rc1 to resolve CVE-2021-3521, CVE-2021-35938 and CVE-2021-35939

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
Patch to backport CVE-2021-35938 and CVE-2021-35939 resulted in RPM failing to install packages, root cause unclear.
Upgrading to 4.18.0-rc1 to pull in CVE fixes instead. A fixup patch is included from upstream resolving two outstanding issues with the release candidate.

###### Change Log  

- Upgrade rpm to 4.18.0-rc1
  - Include 4.18.0-rc1-bugfix.patch to fix bugs in the release candidate
- Resolves CVE-2021-3521 CVE-2021-35938 CVE-2021-35939

###### Does this affect the toolchain?  

**YES**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2021-3521
- https://nvd.nist.gov/vuln/detail/CVE-2021-35938
- https://nvd.nist.gov/vuln/detail/CVE-2021-35939

###### Test Methodology

- Local toolchain build done
- Full pipeline build in progress: 235918

### CODE REVIEW

**Comment by dmcilvaney** [comment]
Updated to include CVE-2021-35939 as well, since its fixed by the same patch set.

**Comment by dmcilvaney** [comment]
Pipeline failed, investigating

**Comment by dmcilvaney** [comment]
Marking as draft, patch needs to be manually applied. May be the better path is to bump to the 4.18.* release candidate.

**Comment by aditjha-msft** [comment]
Hey Daniel,

I did some digging into the failed pipeline, it failed on  mlx-bootctl and no one has touched this package since July 29th.
This is the line that it failed on:
time="2022-09-07T03:59:21Z" level=debug msg="+ make -C /usr/lib/modules/5.15.63.1-1.cm2/build M=/usr/src/mariner/BUILD/mlx-bootctl-1.5/obj/default"
time="2022-09-07T03:59:21Z" level=debug msg="make: *** /usr/lib/modules/5.15.63.1-1.cm2/build: No such file or directory.  Stop."

I feel like this was a random pipeline failure and I'm just going to requeue it. Not sure if you have any specific thoughts about the failed pipeline.

Thanks,
Adit
________________________________

**Comment by dmcilvaney** [comment]
Patch backport did not work as expected, shorter path to functionality was making the jump to the release candidate 4.18.0 build. Pulled in the outstanding bug fixes on top of the release candidate.

**Comment by dmcilvaney** [comment]
Force pushed to resolve manifest conflict with python

**Comment by dmcilvaney** [comment]
Thoughts on how to handle the `rc1` part of version? Redhat seems to favor <version>-0.rc#-<release> so it orders properly with non-rc releases.
This is not passing our release format check however.
Alternative is to drop the `rc1` from the version-release completely?

**Comment by dmcilvaney** [comment]
Running a new build: 235918

File: SPECS/rpm/rpm.spec [hunk 1]
```diff
@@ -1,31 +1,35 @@
...
-Release:        9%{?dist}
+Version:        4.18.0
+Release:        0.rc1.1%{?dist}
 License:        GPLv2+ AND LGPLv2+ AND BSD
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          Applications/System
 URL:            https://rpm.org
-Source0:        https://github.com/rpm-software-management/rpm/archive/%{name}-%{version}-release.tar.gz
+Source0:        https://github.com/rpm-software-management/rpm/archive/%{name}-%{version}-rc1.tar.gz
```
[the remark this answers is not in the document]
Comment (dmcilvaney) [reply SPECS/rpm/rpm.spec:10]: I saw that, didn't want to rock the boat but might be worth doing now.

**Comment by dmcilvaney** [comment]
Pulled a few more commits back into the rc, and updated the source URL. Doing a local build now to validate.

File: SPECS/rpm/rpm.spec (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmcilvaney) [reply SPECS/rpm/rpm.spec]: Updated

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**